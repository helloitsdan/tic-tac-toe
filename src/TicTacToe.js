import React from 'react'
import { Provider } from 'react-redux'

import Game from './modules/Game'

import createStore from './store'

const TicTacToe = () => (
  <Provider store={createStore()}>
    <div className="c-background" />
    <div className="o-wrapper">
      <h1>Tic-tac-toe!</h1>
      <Game />
    </div>
  </Provider>
)

export default TicTacToe
