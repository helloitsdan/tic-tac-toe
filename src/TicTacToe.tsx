import React, { FunctionComponent } from 'react'
import { Provider } from 'react-redux'

import Game from './modules/Game'

import createStore from './store'
import Background from './components/Background'

const TicTacToe: FunctionComponent = () => (
  <Provider store={createStore()}>
    <Background />
    <div className="o-wrapper">
      <Game />
    </div>
  </Provider>
)

export default TicTacToe
