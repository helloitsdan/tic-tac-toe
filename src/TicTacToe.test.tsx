import React from 'react'
import { render } from '@testing-library/react'

import TicTacToe from './TicTacToe'

describe('TicTacToe', () => {
  it('renders', () => {
    const { asFragment } = render(<TicTacToe />)
    expect(asFragment()).toMatchSnapshot()
  })
})