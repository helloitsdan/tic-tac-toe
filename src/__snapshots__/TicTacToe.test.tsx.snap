// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TicTacToe renders 1`] = `
<DocumentFragment>
  <div
    class="c-background"
  />
  <div
    class="o-wrapper"
  >
    <div
      class="o-game-column"
    >
      <h1>
        Tic-tac-toe!
      </h1>
      <div
        class="o-game-columns__ui"
      >
        <div
          class="c-game-board"
        >
          <svg
            class="c-game-board__grid"
          >
            board.svg
          </svg>
          <div
            class="c-game-board__square"
            data-testid="Board/Square"
          >
            <button
              class="c-board-button"
            >
              <span>
                 
              </span>
            </button>
          </div>
          <div
            class="c-game-board__square"
            data-testid="Board/Square"
          >
            <button
              class="c-board-button"
            >
              <span>
                 
              </span>
            </button>
          </div>
          <div
            class="c-game-board__square"
            data-testid="Board/Square"
          >
            <button
              class="c-board-button"
            >
              <span>
                 
              </span>
            </button>
          </div>
          <div
            class="c-game-board__square"
            data-testid="Board/Square"
          >
            <button
              class="c-board-button"
            >
              <span>
                 
              </span>
            </button>
          </div>
          <div
            class="c-game-board__square"
            data-testid="Board/Square"
          >
            <button
              class="c-board-button"
            >
              <span>
                 
              </span>
            </button>
          </div>
          <div
            class="c-game-board__square"
            data-testid="Board/Square"
          >
            <button
              class="c-board-button"
            >
              <span>
                 
              </span>
            </button>
          </div>
          <div
            class="c-game-board__square"
            data-testid="Board/Square"
          >
            <button
              class="c-board-button"
            >
              <span>
                 
              </span>
            </button>
          </div>
          <div
            class="c-game-board__square"
            data-testid="Board/Square"
          >
            <button
              class="c-board-button"
            >
              <span>
                 
              </span>
            </button>
          </div>
          <div
            class="c-game-board__square"
            data-testid="Board/Square"
          >
            <button
              class="c-board-button"
            >
              <span>
                 
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="o-game-columns__menu"
      >
        <div
          class="u-push-bottom"
          data-testid="Status/TurnStatus"
        >
          <div
            data-testid="TurnStatus"
          >
            <span
              data-testid="TurnStatus/Player"
            >
              Player One
            </span>
            's turn!
          </div>
        </div>
        <nav>
          <button
            class="c-button"
          >
            Start new game!
          </button>
        </nav>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
